core/variable_operations.cpp

```diff
@@ -327,10 +327,17 @@ Variable reverse(Variable var, const Dim dim) {
 Variable copy(const VariableConstProxy &var) { return Variable(var); }
 
 /// Merges all masks contained in the MasksConstProxy into a single Variable
-Variable masks_merge(const MasksConstProxy &masks) {
+Variable masks_merge(const MasksConstProxy &masks, const Dim dim,
+                     const std::optional<Slice> &slice) {
```

Comment:
Why push the slicing into this function? Would it not make more sense to do it at the call site?